services/src/main/java/org/keycloak/utils/MediaTypeMatcher.java

```diff
@@ -1,16 +1,24 @@
 package org.keycloak.utils;
 
 import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
 
 public class MediaTypeMatcher {
 
     public static boolean isHtmlRequest(HttpHeaders headers) {
-        for (jakarta.ws.rs.core.MediaType m : headers.getAcceptableMediaTypes()) {
-            if (!m.isWildcardType() && m.isCompatible(jakarta.ws.rs.core.MediaType.TEXT_HTML_TYPE)) {
+        return isAcceptMediaType(headers, MediaType.TEXT_HTML_TYPE);
+    }
+
+    public static boolean isJsonRequest(HttpHeaders headers) {
+        return isAcceptMediaType(headers, MediaType.APPLICATION_JSON_TYPE);
+    }
+
+    private static boolean isAcceptMediaType(HttpHeaders headers, MediaType textHtmlType) {
```

Comment:
minor: renaming the latter variable to `mediaType` might be better (not a blocker for this PR unless there are other points to address).

Reply:
Thanks for the suggestion. I just forgot this one.
I will take care of the param name next time.